File: backend-event-streamer/internal/domain/account/account-created.go
package account

type AccountCreatedEvent struct {
	Email     string `json:"email"`
	FullName  string `json:"fullName"`
	Username  string `json:"username"`
	AccountID string `json:"accountId"`
	CreatedAt string `json:"createdAt"`
}

func HandleAccountCreated(eventData map[string]interface{}) error {
	return nil
}

File: backend-api/src/Domain/BankAccount/Events/AccountCreated.hs
{-# LANGUAGE DeriveAnyClass #-}
{-# LANGUAGE DeriveGeneric #-}
{-# LANGUAGE OverloadedStrings #-}

module Domain.BankAccount.Events.AccountCreated (AccountCreated (..), eventName) where

import Data.Aeson (FromJSON, ToJSON)
import Data.Text (Text)
import Data.Time.Clock (UTCTime)
import Data.UUID (UUID)
import GHC.Generics (Generic)

data AccountCreated = AccountCreated
  { accountId :: UUID,
    username :: Text,
    fullName :: Text,
    email :: Text,
    createdAt :: UTCTime
  }
  deriving (Show, Generic, FromJSON, ToJSON)

eventName :: Text
eventName = "AccountCreated"
